pemFioi/quickpi/blocklyQuickPi_lib.js

```diff
@@ -1597,6 +1597,11 @@ var getContext = function (display, infos, curLevel) {
     // Create a base context
     var context = quickAlgoContext(display, infos);
 
+    // we set the lib involved to Quick-Pi
+    context.title = "Quick-Pi";
+
+    window.conceptViewer.contextTitle = context.title;
```

Comment:
We might have multiple contexts loaded (it's another ongoing project), so this line will cause issues.

In quickAlgo, we generally have all information pulled from the contexts : contexts define "public" functions and properties through the `context` object, and quickAlgo will read the context properties and act accordingly. It's best to keep that logic, especially for cases where there can be multiple contexts.

Here, that means that instead of having the context directly access conceptViewer, it would be better to have quickAlgo pull the title and give it to conceptViewer. That could be done for instance here [at line 95 of subtask.js](https://github.com/France-ioi/bebras-modules/blob/master/pemFioi/quickAlgo/subtask.js#L95), where you would simply add `window.conceptViewer.contextTitle = this.context.title;`.

On another note : be careful about using global variables, as for instance here, conceptViewer might not be loaded (that happens for instance in the auto-generated documentation). So if you were to keep the line here, it's safer to write `if(window.conceptViewer) { window.conceptViewer.contextTitle = context.title; }`. (You won't need to make that check in the section of subtask.js I linked above, as it is only executed when the task says it needs conceptViewer.)

Reply:
I have set window.conceptViewer.contextTitle from subtask.js in the next push